stdlib/io/src/main/java/org/ballerinalang/stdlib/io/nativeimpl/GetTable.java

```diff
@@ -35,69 +58,107 @@
     private GetTable() {
     }
 
-    //TODO Table remove - Fix
+    public static Object getTable(ObjectValue csvChannel, TypedescValue typedescValue, ArrayValue key) {
+        try {
+            final ObjectValue delimitedObj =
+                    (ObjectValue) csvChannel.get(StringUtils.fromString(CSV_CHANNEL_DELIMITED_STRUCT_FIELD));
+            DelimitedRecordChannel delimitedChannel = (DelimitedRecordChannel) delimitedObj
+                    .getNativeData(IOConstants.TXT_RECORD_CHANNEL_NAME);
+            if (delimitedChannel.hasReachedEnd()) {
+                return IOUtils.createEoFError();
+            }
+            List<String[]> records = new ArrayList<>();
+            while (delimitedChannel.hasNext()) {
+                records.add(delimitedChannel.read());
+            }
+            return getTable(typedescValue, key, records);
+        } catch (BallerinaIOException | BallerinaException e) {
+            String msg = "failed to process the delimited file: " + e.getMessage();
+            log.error(msg, e);
```

Comment:
Do we need to log this error here? As we are returning the error, anyway user will have to handle that.

Reply:
Updated in 4312c4f